lib/matplotlib/axis.py

```diff
@@ -781,13 +818,12 @@ def reset_ticks(self):
         # build a few default ticks; grow as necessary later; only
         # define 1 so properties set on ticks will be copied as they
         # grow
-        del self.majorTicks[:]
-        del self.minorTicks[:]
-
-        self.majorTicks.extend([self._get_tick(major=True)])
-        self.minorTicks.extend([self._get_tick(major=False)])
-        self._lastNumMajorTicks = 1
-        self._lastNumMinorTicks = 1
+        if not isinstance(self.majorTicks, LazyDefaultTickList):
+            del self.majorTicks[:]
```

Comment:
You shouldn't need this any more since the list is being removed, no?

Reply:
Guess you're right. I was wondering anyway, why the previous code bothered to del every item from the list and not just created a new list (`self.majorTicks=[]` would have been my expectation in the original code).

Also, can you tell if there's still a need for `self._lastNumMajorTicks`? I haven't looked into this so far. It feels like a helper state, that might not be necessary. If you don't know right now, I can check myself.

Reply:
I believe it worked this way so that any use of `self.majorTicks` would point to the same list. It's interesting that all the tests still pass though, as clearly this new class does not make that same guarantee.